mono/utils/mono-linked-list-set.c

```diff
@@ -126,7 +126,7 @@ mono_lls_find (MonoLinkedListSet *list, MonoThreadHazardPointers *hp, uintptr_t
 				mono_memory_write_barrier ();
 				mono_hazard_pointer_clear (hp, 1);
 				if (list->free_node_func)
-					mono_thread_hazardous_free_or_queue (cur, list->free_node_func, list->locking, context);
+					mono_thread_hazardous_queue_free (cur, list->free_node_func);
 			} else
```

Comment:
This is a conservative assumption about free not being good outside of _remove.

This only happens under extreme load so it's ok to have it.